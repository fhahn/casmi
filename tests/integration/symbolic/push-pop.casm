// symbolic

CASM test

function (symbolic) a: -> List(Int)
function (symbolic) b: -> Int
function (symbolic) c: -> Int

rule main = {|

  push 1 into a
  push 2 into a

  pop b from a
  pop c from a

  program( self ) := undef
|}

init main

