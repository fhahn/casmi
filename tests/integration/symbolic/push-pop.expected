forklog:
tff(symbolNext, type, sym2: $int).
fof(id0,hypothesis,sta(1,sym2)).%CREATE: a
tff(symbolNext, type, sym5: $int).
fof(id1,hypothesis,stb(1,sym5)).%CREATE: b
tff(symbolNext, type, sym7: $int).
fof(id2,hypothesis,stc(1,sym7)).%CREATE: c
tff(symbolNext, type, sym3: $int).
fof(id3,hypothesis,fpush(sym2, 1, sym3)).
tff(symbolNext, type, sym4: $int).
fof(id4,hypothesis,fpush(sym3, 2, sym4)).
tff(symbolNext, type, sym6: $int).
fof(id5,hypothesis,fpop(sym4,2,sym6)).
tff(symbolNext, type, sym8: $int).
fof(id6,hypothesis,fpop(sym6,1,sym8)).
fof(id7,hypothesis,sta(2,sym8)).%UPDATE: a
fof(id8,hypothesis,stb(2,2)).%UPDATE: b
fof(id9,hypothesis,stc(2,1)).%UPDATE: c
fof(final0,hypothesis,sta(0,sym8)).%FINAL: a
fof(final1,hypothesis,stb(0,2)).%FINAL: b
fof(final2,hypothesis,stc(0,1)).%FINAL: c

